source/frontend/properties/gc/Material/StageSurface.cpp

```diff
@@ -177,103 +178,141 @@ void drawProperty(kpi::PropertyDelegate<IGCMaterial>& delegate,
 
     if (ImGui::CollapsingHeader("Color Stage",
                                 ImGuiTreeNodeFlags_DefaultOpen)) {
-      // TODO: Only add for now..
-      if (stage.colorStage.formula == librii::gx::TevColorOp::add) {
-        TevExpression expression(stage.colorStage);
-
-        ImGui::SetWindowFontScale(1.3f);
-        ImGui::Text("%s", expression.getString());
-        ImGui::SetWindowFontScale(1.0f);
-
-        int a = static_cast<int>(stage.colorStage.a);
-        int b = static_cast<int>(stage.colorStage.b);
-        int c = static_cast<int>(stage.colorStage.c);
-        int d = static_cast<int>(stage.colorStage.d);
-        bool clamp = stage.colorStage.clamp;
-        int bias = static_cast<int>(stage.colorStage.bias);
-        int scale = static_cast<int>(stage.colorStage.scale);
-        int dst = static_cast<int>(stage.colorStage.out);
-
-        const auto ksel = DrawKonstSel(stage.colorStage.constantSelection);
-        STAGE_PROP(colorStage.constantSelection, ksel);
-
-        auto draw_color_operand = [&](const char* title, int* op, u32 op_mask) {
-          ConditionalActive g(expression.isUsed(op_mask), false);
-          ImGui::Combo(title, op, colorOpt);
-        };
-
-        draw_color_operand("Operand A", &a, A);
-        draw_color_operand("Operand B", &b, B);
-        draw_color_operand("Operand C", &c, C);
-        draw_color_operand("Operand D", &d, D);
+      TevExpression expression(stage.colorStage);
+
+      int formula = static_cast<int>(stage.colorStage.formula);
+      int op = std::clamp(formula, 0, 2);
+      ImGui::Combo("Operation", &op, "Add\0Subtract\0Compare\0");
+      const bool comp_mode = op >= 2;
+
+      if (comp_mode) {
+        op = std::clamp(formula - 8, 0, 8);
+
+        int chan = op >> 1;
+        int mode = op & 1;
+
+        // TODO: better names?
+        ImGui::Combo("Registers", &chan, "Red\0Green, Red\0Blue, Green, Red (24 bit)\0Blue, Green, Red (8 bit)\0");
+        ImGui::Combo("Mode", &mode, "Equal\0Greater\0");
+
+        op = (chan << 1) | mode + 8;
+      }
```

Comment:
I'd rather we didn't play such bit-twiddling games in the middle of UI code. Error-prone code should be extracted into functions that can be scrutinized in isolation and tested if needed (see librii::hx). Please also refrain from using magic numbers, we have an enum `gx::TevColorOp` for this.